examples/gno.land/r/demo/disperse/disperse.gno

```diff
@@ -0,0 +1,108 @@
+package disperse
+
+import (
+	"std"
+
+	tokens "gno.land/r/demo/grc20factory"
+)
+
+// Get address of Disperse realm
+var realmAddr = std.CurrentRealm().Addr()
+
+// DisperseUgnot parses receivers and amounts and sends out ugnot
+// The function will send out the coins to the addresses and return the leftover coins to the caller
+// if there are any to return
+func DisperseUgnot(addresses []std.Address, coins std.Coins) {
+	coinSent := std.GetOrigSend()
+	caller := std.PrevRealm().Addr()
+	banker := std.GetBanker(std.BankerTypeOrigSend)
+
+	// Check if realm balance is equal to amount sent
+	if banker.GetCoins(realmAddr).AmountOf("ugnot") != coinSent.AmountOf("ugnot") {
+		panic(ErrBalanceNotEqToAmountSent)
+	}
+
+	if len(addresses) != len(coins) {
+		panic(ErrNumAddrValMismatch)
+	}
+
+	for _, coin := range coins {
+		if coin.Amount <= 0 {
+			panic(ErrNegativeCoinAmount)
+		}
+	}
+
+	var totalAmount int64
+	for _, coin := range coins {
+		totalAmount += coin.Amount
+	}
+
+	// Send coins
+	for i, _ := range addresses {
+		banker.SendCoins(realmAddr, addresses[i], std.NewCoins(coins[i]))
+	}
+
+	// Return possible leftover coins
+	for _, coin := range coinSent {
+		leftoverAmt := banker.GetCoins(realmAddr).AmountOf(coin.Denom)
```

Comment:
I can't really do this because of the new changes in https://github.com/gnolang/gno/pull/2613/commits/128a8e87132ac05b4759e8c8369301ea642f9d06
Since I would get a `ErrMismatchBetweenSentAndParams` error because I'm manipulating the coin amounts mid loop